internal/e2e/profiler/profiler.go

```diff
@@ -0,0 +1,98 @@
+/*
+   Copyright 2023 The Kubernetes Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"github.com/stealthrocket/wzprof"
+	"io"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	"log"
+	"os"
+	wasm "sigs.k8s.io/kube-scheduler-wasm-extension/scheduler/plugin"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/scheduler/testdata"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/experimental"
+)
+
+func main() {
+	guestPath := os.Args[1] // must be a debug build
+	node := testdata.NodeReal
+	pod := testdata.PodReal
+	ni := framework.NewNodeInfo()
+	ni.SetNode(node)
+
+	// Configure the profiler, which integrates via context configuration.
+	sampleRate := 1.0
+	cpu := wzprof.NewCPUProfiler()
+	mem := wzprof.NewMemoryProfiler()
+	ctx := context.WithValue(context.Background(),
+		experimental.FunctionListenerFactoryKey{},
+		experimental.MultiFunctionListenerFactory(
+			wzprof.Sample(sampleRate, cpu),
+			wzprof.Sample(sampleRate, mem),
+		),
+	)
+
+	// Pass the profiling context to the plugin.
+	plugin, err := wasm.NewFromConfig(ctx, wasm.PluginConfig{GuestPath: guestPath})
+	if err != nil {
+		log.Fatal("failed to create plugin:", err)
+	}
+	defer plugin.(io.Closer).Close()
+
+	// Build the debug symbols needed in the profiler.
+	symbols := buildSymbolizer(ctx, guestPath)
+
+	// Profile around the Filter function
+	cpu.StartProfile()
+	s := plugin.(framework.FilterPlugin).Filter(ctx, nil, pod, ni)
+	if want, have := framework.Success, s.Code(); want != have {
+		log.Fatal("filter failed:", want, "!=", have, ":", s.Message())
```

Comment:
Please confirm if it's OK to use log.Fatal? That means `defer plugin.(io.Closer).Close()` above won't be called.

Reply:
if you prefer panic that's fine with me. Personally, I'm not concerned with this at all (os.Exit before mmunmap). Main thing is least distractions and it seems log.Fatal distracts a core maintainer. That's good enough reason to switch to panic.

Reply:
switched.. seems better as the stack trace will help anyway